### ISSUE TITLE
Resources in decimal values converting to m

### ISSUE THREAD

**Issue by saiharshitachava** [kubernetes/kubernetes#91156]
We have given a resource limit for memory as 2.7G at the time of deployment.. as below

          resources:
            limits:
              cpu: "1"
              memory: 2.7Gi
            requests:
              cpu: "0.25"
              memory: 512Mi

The values have now been converted into some small I think the millibytes

        resources:
          limits:
            cpu: "1"
            memory: 2899102924800m
          requests:
            cpu: 250m
            memory: 512Mi

Could someone let me know why this is not in Mi and it converted to m?

**Comment by vayado** [comment]
Has this been solved?

**Comment by jakeMurphyAtPropertyMe** [comment]
@vayado I don't think this will be solved, i found this related issue which makes it seem intended due to rounding: https://github.com/kubernetes/kubectl/issues/1426
